cbor/record.go

```diff
@@ -20,31 +20,45 @@ func init() {
 
 // record defines the node structure of a record.
 type record struct {
-	Block cid.Cid
-	Sig   []byte
-	Prev  cid.Cid `refmt:",omitempty"`
+	Block     cid.Cid
+	Sig       []byte
+	AuthorSig []byte
+	Prev      cid.Cid `refmt:",omitempty"`
+}
+
+type CreateRecordConfig struct {
+	Block      format.Node
+	Prev       cid.Cid
+	Key        ic.PrivKey
+	AuthorKey  ic.PrivKey
+	ServiceKey crypto.EncryptionKey
 }
 
 // CreateRecord returns a new record from the given block and log private key.
-func CreateRecord(ctx context.Context, dag format.DAGService, block format.Node, prev cid.Cid, sk ic.PrivKey, key crypto.EncryptionKey) (net.Record, error) {
-	payload := block.Cid().Bytes()
-	if prev.Defined() {
-		payload = append(payload, prev.Bytes()...)
+func CreateRecord(ctx context.Context, dag format.DAGService, config CreateRecordConfig) (net.Record, error) {
+	payload := config.Block.Cid().Bytes()
+	if config.Prev.Defined() {
+		payload = append(payload, config.Prev.Bytes()...)
 	}
-	sig, err := sk.Sign(payload)
+	sig, err := config.Key.Sign(payload)
+	if err != nil {
+		return nil, err
+	}
+	asig, err := config.AuthorKey.Sign(payload)
```

Comment:
One thing to remember here, this can happen on the client when you use `AddRecord`. As I mentioned in the PR description, the host key is used for `CreateRecord`, because in this case, your relying on the remote to create the entire thing. 

And it follows... since the client is doing the authoring, they could use whatever key they want... it just better match the one in the thread ACL. You can imagine a case where the remote has the log private key and not the author's private key.